packages/react-log-viewer/src/LogViewer/loggerToolbar.tsx

```diff
@@ -0,0 +1,177 @@
+
+  /* Defaulting the names of the dropdown items if the default toolbar */
+  useEffect(() => {
+    if (dataSourcesAmount > 1 && dataSourceTitles[currentDataSource] === 'Default') {
+      let titleCounter: number = 0;
+      const newTitles: string[] = [];
+      while (titleCounter < dataSourcesAmount) {
+        newTitles.push(`Data Source ${titleCounter + NUMBER_INDEX_DELTA}`);
+        ++titleCounter;
+      }
+
+      setDataSourceTitles(newTitles);
+    }
+  }, [dataSourcesAmount]);
+
+  /* Clearing out the search input */
+  const handleClear = (): void => {
+    setSearchedInput('');
+    setFoundWordIndex(0);
+    setSearchedWordIndexes([]);
+    setRowInFocus(DEFAULT_FOCUS);
+  };
+
+  /* Moving focus over to next row containing searched word */
+  const handleNextSearchItem = (): void => {
+    const oldIndex = searchedWordIndexes.indexOf(rowInFocus);
+    const adjustedSearchCount = currentSearchItemCount + 1;
+    let temp = foundWordIndex;
+
+    if (oldIndex >= searchedWordIndexes.length - NUMBER_INDEX_DELTA) {
+      return null;
+    }
+
+    setCurrentSearchItemCount(adjustedSearchCount);
+    setFoundWordIndex(++temp);
+    scrollToRow(searchedWordIndexes[foundWordIndex]);
+  };
+
+  /* Moving focus over to next row containing searched word */
+  const handlePrevSearchItem = (): void => {
+    const oldIndex = searchedWordIndexes.indexOf(rowInFocus);
+    const adjustedSearchCount = currentSearchItemCount - 1;
+    let temp = foundWordIndex;
+
+    if (oldIndex < 0) {
+      return null;
+    }
+
+    setCurrentSearchItemCount(adjustedSearchCount);
+    setFoundWordIndex(--temp);
+    scrollToRow(searchedWordIndexes[--temp]);
+  };
+
+  /* Way to make this logic much simpler? */
+  const toolbarItems = (
+    <>
+      <ToolbarItem>
+        <ToolbarDropdown
+          currentDataSource={currentDataSource}
+          setCurrentDataSource={setCurrentDataSource}
+          dataSourcesAmount={dataSourcesAmount}
+          dataSourceTitles={dataSourceTitles}
+          setDataSourceTitles={setDataSourceTitles}
+        />
+      </ToolbarItem>
+      <ToolbarItem>
+        <ToolbarGroup>
+          <SearchInput
+            placeholder="Search"
+            value={searchedInput}
+            onNextClick={() => handleNextSearchItem()}
+            onPreviousClick={() => handlePrevSearchItem()}
+            onClear={() => handleClear()}
+            onChange={input => setSearchedInput(input)}
+            resultsCount={`${currentSearchItemCount} / ${searchedWordIndexes.length}`}
+            className={styles.toolbarSearchbar}
+          />
+        </ToolbarGroup>
+      </ToolbarItem>
+      <ToolbarGroup alignment={{ default: 'alignRight' }}>
+        <ToolbarItem>
+          {includesPlay && (
+            <Button variant="plain" className={css(styles.searchbarBtn)}>
+              <PlayIcon />
+            </Button>
+          )}
+        </ToolbarItem>
+        <ToolbarItem>
+          {includesFullScreen && (
```

Comment:
you should only render the `ToolbarItem` if it's contents are going to be rendered. You don't want empty toolbaritems, but most of these wont be necessary for the initial PR.